cmd/kops/create_cluster.go

```diff
@@ -760,19 +765,34 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 		cluster.Spec.API = &api.AccessSpec{}
 	}
 	if cluster.Spec.API.IsEmpty() {
-		switch cluster.Spec.Topology.Masters {
-		case api.TopologyPublic:
-			cluster.Spec.API.DNS = &api.DNSAccessSpec{}
```

Comment:
I think we didn't expose the api options through flags previously.

Public topologies would default to DNS, Private topologies would default to ELB.

I think if we have conflicting flags - i.e. if we can't even populate the cluster consistently - we should return error immediately.  But if we can populate the cluster, we should use validation to verify the flags make sense, so that no matter how the cluster is populated we perform the same validation.

So in this case, I don't think we have a conflict ... yet.